### PR TITLE
NUTS: detach `potential_energy`'s from the computation graph

### PR DESCRIPTION
**Pull request by jburroni**
`velocity_verlet` is not detaching the `potential_energy` from the computation graph, and so it keeps track of unnecessary computations.
This might not be the right place to add the `detach` action

### CODE REVIEW

File: pyro/ops/integrator.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def velocity_verlet(z, r, potential_fn, inverse_mass_matrix, step_size, num_step
                                                                         inverse_mass_matrix,
                                                                         step_size,
                                                                         z_grads)
-    return z_next, r_next, z_grads, potential_energy
```
Comment (neerajprad) [inline pyro/ops/integrator.py]: Could you share your model where this happens? At no point in HMC do we set `requires_grad = True`, except for when we take the gradient `_potential_grad` below, right after which this is set to False again. Are you by any chance using `pyro.param` in the model?

**Comment by jburroni** [comment]
```python
import torch
from torch import tensor, rand
import pyro
import torch.distributions.constraints as constraints
import pyro.distributions as dist
from pyro.infer import mcmc


def model(x: 'int[10]'=None):
    theta: 'real' = pyro.sample('theta', dist.Uniform(0.0, 1.0))
    pyro.sample('theta' + '__1', dist.Beta(1, 1), obs=theta)
    for i in range(1, 10 + 1):
        pyro.sample('x' + '__{}'.format(i - 1) + '__2', dist.Bernoulli(
            theta), obs=x[i - 1])

x = torch.Tensor([0, 0, 0, 0, 0, 0, 1, 0, 0, 1])
def nuts(model, **kwargs):
    nuts_kernel = mcmc.NUTS(model)
    return mcmc.MCMC(nuts_kernel, **kwargs)

nuts(model, num_samples=10000, warmup_steps=100).run(x=x)
```

File: pyro/ops/integrator.py (hunk 1 above)
Comment (neerajprad) [reply pyro/ops/integrator.py]: Okay I see the issue. We are also returning `potential_energy` from `_potential_grad`. Could you instead make this change in `_potential_grad` below?

```python
return dict(zip(z_keys, grads)), potential_energy.detach()
```

**Comment by fehiepsi** [comment]
Thanks for detecting this! It seems like an issue to me. I just have a minor comment: Could you detach at _potential_grad function instead? :)

File: pyro/ops/integrator.py (hunk 1 above)
Comment (jburroni) [reply pyro/ops/integrator.py]: done
